### ISSUE TITLE
Error message: ERROR: Initial value and slope must be finite when using method "ar"

### ISSUE THREAD

**Issue by chriscandelaria** [matthieugomez/SparseFactorModels.jl#1]
I received the following error after submitting an IFE model on Julia using the SparseFactorModels package:
`ERROR: Initial value and slope must be finite`
` in hz_linesearch! at /home/candelar/.julia/v0.3/Optim/src/linesearch/hz_linesearch.jl:184`
` in hz_linesearch! at /home/candelar/.julia/v0.3/Optim/src/linesearch/hz_linesearch.jl:176 (repeats 9 times)`
` in update_half! at /home/candelar/.julia/v0.3/SparseFactorModels/src/algorithms/gd.jl:88`
` in update! at /home/candelar/.julia/v0.3/SparseFactorModels/src/algorithms/gd.jl:59`
` in fit! at /home/candelar/.julia/v0.3/SparseFactorModels/src/algorithms/gd.jl:135`
` in fit at /home/candelar/.julia/v0.3/SparseFactorModels/src/fit.jl:123`
` in include at ./boot.jl:245`
` in include_from_node1 at ./loading.jl:128`
`while loading /share/PI/reininge/Users/candelar/SchFinIFE/julia/juliacode1.jl, in expression starting on line 9`

Here are the commands I submitted to run the IFE model:

`# Load packages`
`using SparseFactorModels, DataFrames, FixedEffectModels`
`# Import data`
`df = readtable("juliaexportTEMP.csv")`
`# Prepare data for SparseFactorModels`
`df[:pdistid] = pool(df[:distid])`
`df[:pfrplyear] = pool(df[:frplyear])`
`df[:pyear] = pool(df[:year])`
`df[:pstate] = pool(df[:stabb_num])`
`df[:pstatefrpl] = pool(df[:stabbfrpl])`
`# Factor structure`
`fmtestgradrate01_tc11 = SparseFactorModel(:pstate, :pyear, 1)`

`# Estimation Model: The fit function (below) appears on line 9 of my program file. This is the line to which the error above refers.`
`fmtestgradrate01_tc11RUN = fit(fmtestgradrate01_tc11, gradrate01_tc ~  TECfrpl1yr1 + TECfrpl1yr2 + TECfrpl1yr3 + TECfrpl1yr4 + TECfrpl1yr5 + TECfrpl1yr6 + TECfrpl1yr7 + TECfrpl1yr8 + TECfrpl1yr9 + TECfrpl1yr10 + TECfrpl1yr11 + TECfrpl1yr12 + TECfrpl1yr13 + TECfrpl1yr14 + TECfrpl1yr15 + TECfrpl1yr16 + TECfrpl1yr17 + TECfrpl1yr18 + TECfrpl1yr19on + TECfrpl2yr1 + TECfrpl2yr2 + TECfrpl2yr3 + TECfrpl2yr4 + TECfrpl2yr5 + TECfrpl2yr6 + TECfrpl2yr7 + TECfrpl2yr8 + TECfrpl2yr9 + TECfrpl2yr10 + TECfrpl2yr11 + TECfrpl2yr12 + TECfrpl2yr13 + TECfrpl2yr14 + TECfrpl2yr15 + TECfrpl2yr16 + TECfrpl2yr17 + TECfrpl2yr18 + TECfrpl2yr19on + TECfrpl3yr1 + TECfrpl3yr2 + TECfrpl3yr3 + TECfrpl3yr4 + TECfrpl3yr5 + TECfrpl3yr6 + TECfrpl3yr7 + TECfrpl3yr8 + TECfrpl3yr9 + TECfrpl3yr10 + TECfrpl3yr11 + TECfrpl3yr12 + TECfrpl3yr13 + TECfrpl3yr14 + TECfrpl3yr15 + TECfrpl3yr16 + TECfrpl3yr17 + TECfrpl3yr18 + TECfrpl3yr19on + TECfrpl4yr1 + TECfrpl4yr2 + TECfrpl4yr3 + TECfrpl4yr4 + TECfrpl4yr5 + TECfrpl4yr6 + TECfrpl4yr7 + TECfrpl4yr8 + TECfrpl4yr9 + TECfrpl4yr10 + TECfrpl4yr11 + TECfrpl4yr12 + TECfrpl4yr13 + TECfrpl4yr14 + TECfrpl4yr15 + TECfrpl4yr16 + TECfrpl4yr17 + TECfrpl4yr18 + TECfrpl4yr19on   |> pdistid + pfrplyear , df, method=:ar, VcovCluster(:pstate), tol=1e-7, maxiter=80000)`

On August 18, I was able to run the model, but when I recently installed the package updates on my server computer, I started getting this error. For some reason, the `gd` algorithm is being called, even though I specify the use of method `ar`. Perhaps I am missing something? Should `gd` be called? Any help would be greatly appreciated. I am happy to provide any additional information. Thanks!

Note: I am using Julia Language 0.3.1 on a Linux environment. I will also try this on MacOSX.

**Comment by matthieugomez** [comment]
I'm not sure why this happens. It does not happen on the dataset I'm testing the package with. Can you reproduce it on some publicly available dataset with a simple model?

For now, I've just removed the initialization of the model through `:gd` so that should work, but I'm curious to understand why this happens.

To update the package, do `Pkg.checkout("SparseFactorModels.jl")

PS:  instead of adding all dummies for TECfrp1yr, you could just do
```julia
df[:pTECfrpl1yr] = pool(TECfrpl1yr)
df[:pTECfrpl2yr] = pool(TECfrpl2yr)
df[:pTECfrpl3yr] = pool(TECfrpl3yr)
df[:pTECfrpl4yr] = pool(TECfrpl4yr)
 fit(fmtestgradrate01_tc11, gradrate01_tc ~ pTECfrpl1yr + pTECfrpl2yr + pTECfrpl3yr + pTECfrpl4yr)
```
if `TECfrpl1yr` is a categorical variable with the lagged year.
In other words, one variable is `pooled`, it is automatically understood as a categorical variable, and you'll get the same regression than when you add all the dummies

**Comment by chriscandelaria** [comment]
Quick update. SparseFactorModels.jl works well on my MacBook Pro running Yosemite. I think the issue raised above is related to some updates that were being done on the Linux cluster I was using. I also tested SparseFactorModels.jl on a desktop computer that runs Linux, and I am not receiving any errors.

**Comment by matthieugomez** [comment]
Great
